helix-core/src/main/java/org/apache/helix/model/Message.java

```diff
@@ -522,12 +538,25 @@ public long getExecuteStartTimeStamp() {
 
   /**
    * Get the time that this message was created
-   * @return UNIX timestamp
+   * @return UNIX epoch timestamp
    */
   public long getCreateTimeStamp() {
     return _record.getLongField(Attributes.CREATE_TIMESTAMP.toString(), 0L);
   }
 
+  /**
+   * Get the time that the message was expected to be completed
+   * @return UNIX epoch timestamp
+   */
+  public long getCompletionDueTimeStamp() {
+    long completionDue = _record.getLongField(Attributes.COMPLETION_DUE_TIMESTAMP.name(), 0L);
+    if (completionDue == 0) {
+      completionDue = getCreateTimeStamp() + MESSAGE_EXPECT_COMPLETION_PERIOD;
+    }
```

Comment:
We can combine it as:

long completionDue = _record.getLongField(Attributes.COMPLETION_DUE_TIMESTAMP.name(), getCreateTimeStamp() + MESSAGE_EXPECT_COMPLETION_PERIOD);

Reply:
I considered this, but think to avoid calling getCreateTimeStamp() everytime since it involves string passing, may not be big deal though.